### PR TITLE
Null value and label size

### PR DESCRIPTION
**Pull request by VargBurz**
This PR adds several options.

## Changes:
   - Null Value option: map null value if option is set, or throw error
   - Value size option: set value label font size
   - Title size option: set title label font size

image

image

### CODE REVIEW

File: src/mapper.ts [hunk 1]
```diff
@@ -64,19 +65,35 @@ export class Mapper {
           };
           title = this._templateSrv.replace(alias, scopedVars);
         }
+        const values = row.filter((value, idx) => !_.includes(skipIndexes, idx));
+        const mappedValues = this._mapNullValues(values, nullMapping);
```
Comment (rozetko) [inline src/mapper.ts:69]: I guess, we should also use the mapped values earlier, when counting the max value. Can't add a comment there, it's line 52 in this file

File: src/mapper.ts [hunk 2]
```diff
@@ -64,19 +65,35 @@ export class Mapper {
...
+          mappedValues,
           totalMaxValue as number
         )
       }
     );
 
   }
 
+  private _mapNullValues(values: (number | null)[], nullMapping: number | undefined): number[] {
+    const mappedValues = values.map(value => {
```
Comment (rozetko) [inline src/mapper.ts:84]:
```suggestion
    return values.map(value => {
```

File: src/mapper.ts [hunk 3]
```diff
@@ -64,19 +65,35 @@ export class Mapper {
...
+      if(value == null) {
```
Comment (rozetko) [inline src/mapper.ts:85]: I'd make it this way:
```js
if(value !== null) {
  return value;
}
...
```

to reduce nesting

File: src/progress_bar.ts [hunk 4]
```diff
@@ -100,6 +100,22 @@ export class ProgressBar {
     );
   }
 
+  get valueFontSize(): number {
+    const configSize = this._panelConfig.getValue('valueSize');
+    if(configSize === undefined || configSize === null) {
```
Comment (rozetko) [inline src/progress_bar.ts:105]: can it be `null`?

Comment (VargBurz) [reply src/progress_bar.ts:105]: it can be, if you delete value

**Pull request merged**